airflow/providers/amazon/aws/hooks/ec2.py

```diff
@@ -0,0 +1,98 @@
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+import time
+from typing import Optional
+
+from airflow.providers.amazon.aws.hooks.base_aws import AwsBaseHook
+
+
+class EC2Hook(AwsBaseHook):
+    """
+    Interact with AWS EC2 Service.
+    """
+
+    def __init__(self,
+                 region_name: Optional[str] = None,
+                 *args,
+                 **kwargs):
+        self.region_name = region_name
+        self.conn = None
+        super().__init__(*args, **kwargs)
+
+    def get_conn(self):
+        """
+        Return self.conn, if it is None initialize ec2 resource object.
+
+        :return: ec2 resource
+        :rtype: boto3.resource
+        """
+        # set self.conn in first call
+        if not self.conn:
+            self.conn = self.get_resource_type("ec2", self.region_name)
+        return self.conn
+
+    def get_instance(self, instance_id: str):
+        """
+        Get EC2 instance by id and return it.
+
+        :param instance_id: id of the AWS EC2 instance
+        :type instance_id: str
+        :return: Instance object
+        :rtype: ec2.Instance
+        """
+        return self.get_conn().Instance(id=instance_id)
+
+    def get_instance_state(self, instance_id: str) -> str:
+        """
+        Get EC2 instance state by id and return it.
+
+        :param instance_id: id of the AWS EC2 instance
+        :type instance_id: str
+        :return: current state of the instance
+        :rtype: str
+        """
+        return self.get_instance(instance_id=instance_id).state["Name"]
+
+    def wait_for_state(self,
+                       instance_id: str,
+                       target_state: str,
+                       check_interval: float) -> None:
+        """
+        Wait EC2 instance until its state is equal to the target_state.
+
+        :param instance_id: id of the AWS EC2 instance
+        :type instance_id: str
+        :param target_state: target state of instance
+        :type target_state: str
+        :param check_interval: time in seconds that the job should wait in
+            between each instance state checks until operation is completed
+        :type check_interval: float
+        :return: None
+        :rtype: None
+        """
+        instance_state = self.get_instance_state(
+            instance_id=instance_id
+        )
+        while instance_state != target_state:
+            self.log.info("instance state: %s", instance_state)
+            time.sleep(check_interval)
+            instance_state = self.get_instance_state(
+                instance_id=instance_id
+            )
```

Comment:
Not sure, I heard that while True is a bad practice since it's hard to follow the loop condition, but the other one has some code duplication. So, I am not sure, what do you think about this, do you know any post related to this situation. I am facing this situation frequently and still not know what to do :(
[example question frrom stackoverflow](https://stackoverflow.com/questions/743164/emulate-a-do-while-loop-in-python)

Reply:
I think both options are fine (in this case) but my pref would be the second. You can keep yours. LGTM too :)